File: ResourceManager/ViewModels/MainViewModel.vb
Imports VBNetCore.Mvvm.Abstractions
Imports VBNetCore.Mvvm.ViewModels
Imports VBNetCore.Security
Imports System.Threading.Tasks
Imports Microsoft.Extensions.DependencyInjection

Namespace ResourceManager.ViewModels
    Public Class MainViewModel
        Inherits HasPopupViewModel
        Implements IOnLoadedHandler, IPopupable
#Region "IOnLoadedHandler"
        Public Function OnLoadedAsync() As Task Implements IOnLoadedHandler.OnLoadedAsync
            RefreshUI()
            Return Task.FromResult(True)
        End Function
#End Region
        Public Sub RefreshUI()
            If AuthManager.User.Identity.IsAuthenticated Then
                CurrentPage = MainPage
            Else
                CurrentPage = LoginPage
            End If
        End Sub

        Friend Sub GoHome()
            CurrentPage = MainPage
        End Sub

        Public Property CurrentPage As ViewModelBase
            Get
                Return GetProperty(Of ViewModelBase)()
            End Get
            Set(ByVal value As ViewModelBase)
                SetProperty(value)
            End Set
        End Property
        Private ReadOnly Property LoginPage As ViewModelBase
            Get
                Return Application.ServiceProvider.GetRequiredService(Of LoginPageViewModel)()
            End Get
        End Property
        Private ReadOnly Property MainPage As ViewModelBase
            Get
                Return Application.ServiceProvider.GetRequiredService(Of MainPageViewModel)()
            End Get
        End Property
    End Class
End Namespace

File: ResourceManager/ViewModels/MainViewModel.cs
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Options;
using NetCore.Mvvm.Abstractions;
using NetCore.Mvvm.ViewModels;
using NetCore.Security;
using ResourceManager.Services.Abstractions;
using ResourceManager.Settings;
using System;
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Threading;

namespace ResourceManager.ViewModels
{
    public class MainViewModel : HasPopupViewModel, IOnLoadedHandler, IPopupable
    {
        #region IOnLoadedHandler
        public Task OnLoadedAsync()
        {
            RefreshUI();
            return Task.FromResult(true);
        }
        #endregion
        public void RefreshUI()
        {
            if (AuthManager.User.Identity.IsAuthenticated)
                CurrentPage = MainPage;
            else
                CurrentPage = LoginPage;
        }
        
        internal void GoHome()
        {
            CurrentPage = MainPage;
        }

        public ViewModelBase? CurrentPage
        {
            get => GetProperty<ViewModelBase?>();
            set => SetProperty(value);
        }
        private ViewModelBase LoginPage => App.ServiceProvider.GetRequiredService<LoginPageViewModel>();
        private ViewModelBase MainPage => App.ServiceProvider.GetRequiredService<MainPageViewModel>();
    }
}
